source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -6,8 +6,13 @@ namespace Clusters {
 namespace Redis {
 
 bool ClusterSlot::operator==(const Envoy::Extensions::Clusters::Redis::ClusterSlot& rhs) const {
-  return start_ == rhs.start_ && end_ == rhs.end_ && primary_ == rhs.primary_ &&
-         replicas_ == rhs.replicas_;
+  if (start_ != rhs.start_ || end_ != rhs.end_ || *primary_ != *rhs.primary_ ||
+      replicas_.size() != rhs.replicas_.size()) {
+    return false;
+  }
+
+  return std::equal(replicas_.begin(), replicas_.end(), rhs.replicas_.begin(), rhs.replicas_.end(),
+                    [](const auto& it1, const auto& it2) { return it1.first == it2.first; });
```

Comment:
This might have been covered in PR discussions already but how does this differ from just doing `replicas_ = rhs.replicas`?

Reply:
The value type is shared_ptr, and the shared_ptr is not name one even for same ip:port.

Reply:
Oh I see this is just comparing the key. Perhaps add a comment to make this clearer?

Reply:
Ping on this ^^

Would be great to have this covered by a comment

Reply:
Sorry, I forgot that, now I have add a comment. Thank you!